docs/source/cpp/tables.rst

```diff
@@ -77,6 +77,17 @@ has a schema which must match its arrays' datatypes.
 Record batches are a convenient unit of work for various serialization
 and computation functions, possibly incremental.
 
+.. image:: tables-versus-record-batches.svg
+   :alt: A graphical representation of an Arrow Table and a Record Batch, with
+         structure as described in text above.
+
+Because record batches can be represented as a struct array, they can be 
+exported through the C data interface between implementations. Tables and 
+chunked arrays, on the other hand, are concepts in the C++ implementation, not 
+in the Arrow format itself, so they aren't directly portable.
+
+However, a table can be converted to and built from a sequence of record 
+batches easily without needing to copy the underlying array buffers.
```

Comment:
If the columns have different chunking, wouldn't it be necessary to normalize when converting to a RecordBatch?

That being said, a Table is basically a structural view of a RecordBatch: each RecordBatch Array can be wrapped in a ChunkedArray trivially because the buffers are separate and a ChunkedArray is *mostly* an ArrayVector (though, I assume not entirely).

Reply:
> If the columns have different chunking, wouldn't it be necessary to normalize when converting to a RecordBatch?

`TableBatchReader` will create a RB boundary any place there is a chunk boundary. So if you have lots of different chunking you will get lots of uneven batches. I imagine there's some function to re-chunk more reasonably if needed.

Reply:
We don't have any function to re-chunk more reasonably, no. I think heterogeneous chunkings should be uncommon in general, anyway, since all data sources should yield homogenous chunkings (i.e. a stream of RecordBatch).

Reply:
What exactly happens, if you break at every boundary? Does each RecordBatch just store pointers to start and end, and the breaking each boundary ensures contiguity?

Reply:
> Does each RecordBatch just store pointers to start and end, and the breaking each boundary ensures contiguity?

Arrays generally have a `shared_ptr<Buffer>`, an offset, and a length. You can `Slice()` any array, and it just copies the shared pointer to the buffer, and adjusts the offset and length accordingly.